builtin-functions/kphp_internal.txt

```diff
@@ -5,3 +5,32 @@
 // Only KPHP compiler can insert calls to these functions.
 
 function _exception_set_location($e, string $filename, int $line): ^1;
+
+// microtime(false) specialization
+function _microtime_string(): string;
+// microtime(true) specialization
+function _microtime_float(): float;
+
+// hrtime(true) specialization
+function _hrtime_int(): int;
+// hrtime(false) specialization
+function _hrtime_array(): int[];
+
+/** @kphp-internal-result-indexing */
+function _explode_nth($delimiter ::: string, $str ::: string, int $index): string;
+
+/** @kphp-internal-result-indexing */
+function _explode_1($delimiter ::: string, $str ::: string): string;
+
+/** @kphp-internal-result-array2tuple */
+function _explode_tuple2($delimiter ::: string, $str ::: string, int $mask, $limit ::: int = 2+1): tuple(string, string);
+/** @kphp-internal-result-array2tuple */
+function _explode_tuple3($delimiter ::: string, $str ::: string, int $mask, $limit ::: int = 3+1): tuple(string, string, string);
+/** @kphp-internal-result-array2tuple */
+function _explode_tuple4($delimiter ::: string, $str ::: string, int $mask, $limit ::: int = 4+1): tuple(string, string, string, string);
+
+/** @kphp-internal-param-readonly $str */
+function _tmp_substr($str ::: string, $start ::: int, $length ::: int = PHP_INT_MAX): _tmp_string;
```

Comment:
What I don't like is that when you search "_tmp_substr" in a project, it's the only place in compiler where you find it. That's why it seems useless, cause you don't see where calls to this function are constructed. It's very unobvious that a call is created by concatenating "_tmp_" prefix with an original function name.

Could we think of how to avoid this? Probably, `has_tmp_string_specialization()` should look not like a set, but like a map of
```
"substr" -> "_tmp_substr"
```
etc., to drop this magic.